packages/wekan-oidc/oidc_server.js

```diff
@@ -1,349 +1,260 @@
-    var tokenContent = getTokenContent(accessToken);
-    var fields = _.pick(tokenContent, getConfiguration().idTokenWhitelistFields);
-    _.extend(serviceData, fields);
-  }
-
-  if (token.refresh_token)
-    serviceData.refreshToken = token.refresh_token;
-  if (debug) console.log('XXX: serviceData:', serviceData);
-
-  profile.name = userinfo[process.env.OAUTH2_FULLNAME_MAP]; // || userinfo["displayName"];
-  profile.email = userinfo[process.env.OAUTH2_EMAIL_MAP]; // || userinfo["email"];
-
-  if (process.env.OAUTH2_B2C_ENABLED  === 'true'  || process.env.OAUTH2_B2C_ENABLED  === true) {
-    profile.email = userinfo["emails"][0];
-  }
-
-  if (debug) console.log('XXX: profile:', profile);
-
-
-  //temporarily store data from oidc in user.services.oidc.groups to update groups
-  serviceData.groups = (userinfo["groups"] && userinfo["wekanGroups"]) ? userinfo["wekanGroups"] : userinfo["groups"];
-  // groups arriving as array of strings indicate there is no scope set in oidc privider
-  // to assign teams and keep admin privileges
-  // data needs to be treated  differently.
-  // use case: in oidc provider no scope is set, hence no group attributes.
-  //    therefore: keep admin privileges for wekan as before
-  if(Array.isArray(serviceData.groups) && serviceData.groups.length && typeof serviceData.groups[0] === "string" )
-  {
-    user = Meteor.users.findOne({'_id':  serviceData.id});
-
-    serviceData.groups.forEach(function(groupName, i)
-    {
-      if(user?.isAdmin && i == 0)
-      {
-        // keep information of user.isAdmin since in loginHandler the user will // be updated regarding group admin privileges provided via oidc
-        serviceData.groups[i] = {"isAdmin": true};
-        serviceData.groups[i]["displayName"]= groupName;
-      }
-      else
-      {
-        serviceData.groups[i] = {"displayName": groupName};
-      }
-    });
-  }
-
-  // Fix OIDC login loop for integer user ID. Thanks to danielkaiser.
-  // https://github.com/wekan/wekan/issues/4795
-  Meteor.call('groupRoutineOnLogin',serviceData, ""+serviceData.id);
-  Meteor.call('boardRoutineOnLogin',serviceData, ""+serviceData.id);
-
-  return {
-    serviceData: serviceData,
-    options: { profile: profile }
-  };
-});
 
-var userAgent = "Meteor";
-if (Meteor.release) {
-  userAgent += "/" + Meteor.release;
-}
+    if (userinfo.ocs) userinfo = userinfo.ocs.data;
+    if (userinfo.metadata) userinfo = userinfo.metadata;
+    if (debug) console.log('XXX: userinfo:', userinfo);
+
+    serviceData.id = userinfo[process.env.OAUTH2_ID_MAP];
+    serviceData.username = userinfo[process.env.OAUTH2_USERNAME_MAP];
+    serviceData.fullname = userinfo[process.env.OAUTH2_FULLNAME_MAP];
+    serviceData.accessToken = accessToken;
+    serviceData.expiresAt = expiresAt;
+
+    // Oracle OIM and B2C checks remain the same
+    if (process.env.ORACLE_OIM_ENABLED === 'true' || process.env.ORACLE_OIM_ENABLED === true) {
+        if (userinfo[process.env.OAUTH2_EMAIL_MAP]) {
```

Comment:
This can be a `x ? a : b` expression:
```js
serviceData.email = userinfo[userinfo[process.env.OAUTH2_EMAIL_MAP] ? process.env.OAUTH2_EMAIL_MAP : process.env.OAUTH2_USERNAME_MAP]
```